File: Set1_1000/1700_Number_Of_Students_Unable_To_Eat_Lunch.java
class Solution {
    public int countStudents(int[] students, int[] sandwiches) {
        // store the count of 0s and 1s in an array of size 2
        int [] countThem = new int [2];
        for (int number : students)
            countThem[number] += 1;
        for (int i=0; i<sandwiches.length; i++) {
            if (--countThem[sandwiches[i]] < 0)
                return sandwiches.length - i;
        }
        return 0;
    }
}

File: Set1_1000/1700_Number_Of_Students_Unable_To_Eat_Lunch.py
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        queue = collections.deque(students)
        for sandwich in sandwiches:
            if sandwich in queue:
                while queue[0] != sandwich:
                    front = queue.popleft()
                    queue.append(front)
                queue.popleft()
            else:
                return len(queue)
        return 0
            
